packages/app/components/stack-modal/StackOrdersTable.tsx

```diff
@@ -0,0 +1,147 @@
+          </TableRow>
+        </TableHeader>
+        {cowOrders.length > 0 ? (
+          <>
+            <TableCowBody cowOrders={cowOrders} />
+            {hasMoreOrders && (
+              <TableCaption className="pb-2 mt-2">
+                <div
+                  className="text-sm cursor-pointer text-primary-700 hover:underline hover:underline-offset-2"
+                  onClick={addMoreOrders}
+                >
+                  Show more orders
+                </div>
+              </TableCaption>
+            )}
+          </>
+        ) : (
+          <FailedToFetchCowData />
+        )}
+      </Table>
+    </div>
+  );
+};
+
+const TableCowBody = ({ cowOrders }: { cowOrders: CowOrder[] }) => {
+  const formatAmount = (amount: string) =>
+    convertedAmount(amount, 18).toFixed(4);
+  const averagePrice = (cowOrder: CowOrder) =>
+    (
+      Number(cowOrder.executedSellAmount) / Number(cowOrder.executedBuyAmount)
+    ).toFixed(4);
+
+  return (
+    <TableBody>
+      {cowOrders.map(cowOrder => (
+        <TableRow key={cowOrder.uid}>
+          <TableCell className="py-2 md:table-cell">
+            <BodyText
+              size={1}
+              className="text-primary-700 hover:underline hover:underline-offset-2"
+            >
+              <Link
+                target="_blank"
+                href={cowExplorerUrl({
+                  chainId: 100,
+                  uid: cowOrder.uid
+                })}
+              >
+                {addressShortner(cowOrder.uid)}
+              </Link>
+            </BodyText>
+          </TableCell>
+          <TableCell className="py-2">
+            <BodyText className="text-em-med" size={1}>
+              {formatDate(cowOrder.creationDate)}
+            </BodyText>
+          </TableCell>
+          <TableCell className="py-2 text-right ">
+            <BodyText className="text-em-med" size={1}>
+              {formatAmount(cowOrder.executedSellAmount)}
+            </BodyText>
+          </TableCell>
+          <TableCell className="py-2 text-right ">
+            <BodyText className="text-em-med" size={1}>
+              {formatAmount(cowOrder.executedBuyAmount)}
+            </BodyText>
+          </TableCell>
+          <TableCell className="hidden py-2 text-right md:table-cell">
+            {cowOrder.status === "fulfilled" ? (
+              <BodyText className="text-em-med" size={1}>
+                {averagePrice(cowOrder)}
+              </BodyText>
+            ) : (
+              <BodyText className="text-gray-400 animate-pulse" size={1}>
+                fulfilling
+              </BodyText>
+            )}
+          </TableCell>
+        </TableRow>
+      ))}
+    </TableBody>
+  );
+};
+
+const FailedToFetchCowData = () => (
+  <TableCaption className="pb-2 mt-2 space-y-1">
+    <Icon name="warning" className="text-danger-500" />
+    <BodyText weight="medium" size={1} className="text-danger-500">
+      Failed to fetch data from cow api
```

Comment:
We don't need to reference cow api here. "Failed to fetch data." is enough (with punctuation)

Reply:
True. Thanks